### PR TITLE
apply filter on task field from project field and set project field o…

### PR DESCRIPTION
**Pull request by lajidey-foss**
…n task pick if project is empty. fix for issue #19

### CODE REVIEW

**Comment by wojosc** [comment]
@lajidey-foss please add a gif of the result. That would help me to quickly see if the requirement is met. (For the non-developers in our team 🫣🤗

File: project_addon/project_addon/doctype/timesheet_record/timesheet_record.js [hunk 1]
```diff
@@ -13,9 +13,36 @@ frappe.ui.form.on('Timesheet Record', {
 			frm.set_value("from_time", frappe.datetime.now_datetime());
 			frm.refresh_field("from_time");
 		}
+		
+	},
+
+	project: function(frm) {
+		//apply filter to task field on project field selected
+		cur_frm.set_query("task", () => {
```
Comment (KanchanChauhan) [inline project_addon/project_addon/doctype/timesheet_record/timesheet_record.js]: No need to use cur_frm, use only frm. cur_frm is deprecated and exists only in legacy code. Please replace cur_frm with frm everywhere.

File: project_addon/project_addon/doctype/timesheet_record/timesheet_record.js [hunk 2]
```diff
@@ -13,9 +13,36 @@ frappe.ui.form.on('Timesheet Record', {
...
+            return {
+                filters: [
+                    ["Task", "project", "=", cur_frm.doc.project]
+                ]
+            }
+        })
+		cur_frm.refresh_fields(["project", "task"])
```
Comment (KanchanChauhan) [inline project_addon/project_addon/doctype/timesheet_record/timesheet_record.js]: When you set filters, no need for refresh_fields, refresh fields is used when you set the value of the fields.

File: project_addon/project_addon/doctype/timesheet_record/timesheet_record.js [hunk 3]
```diff
@@ -13,9 +13,36 @@ frappe.ui.form.on('Timesheet Record', {
...
 	},
-	
 
+	task: function(frm) {
+		//set project if task is clicked first
+		if(cur_frm.doc.project == null){
```
Comment (KanchanChauhan) [inline project_addon/project_addon/doctype/timesheet_record/timesheet_record.js]: Use if(!frm.doc.project)

File: project_addon/project_addon/doctype/timesheet_record/timesheet_record.js [hunk 4]
```diff
@@ -13,9 +13,36 @@ frappe.ui.form.on('Timesheet Record', {
...
+			frappe.db.get_value('Task', cur_frm.doc.task, 'project')
+				.then(r => {
+					let proj = r.message.project;
+					cur_frm.set_value("project",proj);
+
+					cur_frm.refresh_fields(["task","project"])
```
Comment (KanchanChauhan) [inline project_addon/project_addon/doctype/timesheet_record/timesheet_record.js]: No need to refresh

File: project_addon/project_addon/doctype/timesheet_record/timesheet_record.js [hunk 5]
```diff
@@ -13,9 +13,36 @@ frappe.ui.form.on('Timesheet Record', {
...
 	},
-	
 
+	task: function(frm) {
+		//set project if task is clicked first
+		if(cur_frm.doc.project == null){
+			frappe.db.get_value('Task', cur_frm.doc.task, 'project')
+				.then(r => {
+					let proj = r.message.project;
+					cur_frm.set_value("project",proj);
```
Comment (KanchanChauhan) [inline project_addon/project_addon/doctype/timesheet_record/timesheet_record.js]: This is correct but can be done in fewer lines:
frappe.db.get_value('Task', frm.doc.task, 'project', (r) => {
		        cur_frm.set_value("project",r.project);
		 });

**Comment by lajidey-foss** [comment]
For Tasks field filter

https://github.com/phamos-eu/Project-Addon/assets/85407202/e52fc30c-fe2d-44ef-a054-95ee46fe403c

**Comment by lajidey-foss** [comment]
Project field setter from task selected

https://github.com/phamos-eu/Project-Addon/assets/85407202/82cc5246-46c1-4e2b-9038-37e5cec3df36

**Comment by lajidey-foss** [comment]
All comment has been taken and implemented here
https://github.com/phamos-eu/Project-Addon/pull/21 #21

**Comment by lajidey-foss** [comment]
To keep all conversation in a single place, all correction will be effected on this pr

File: project_addon/project_addon/doctype/timesheet_record/timesheet_record.js [hunk 6]
```diff
@@ -18,30 +18,23 @@ frappe.ui.form.on('Timesheet Record', {
...
 	task: function(frm) {
 		//set project if task is clicked first
-		if(cur_frm.doc.project == null){
-			frappe.db.get_value('Task', cur_frm.doc.task, 'project')
-				.then(r => {
-					let proj = r.message.project;
-					cur_frm.set_value("project",proj);
+		if(!frm.doc.project){
+				frappe.db.get_value('Task', frm.doc.task, 'project', (r) => {
+					cur_frm.set_value("project",r.project);
```
Comment (wojosc) [inline project_addon/project_addon/doctype/timesheet_record/timesheet_record.js]: why still use cur_frm here rather than frm?

**Review by wojosc — APPROVED** [review]

**Comment by wojosc** [comment]
@lajidey-foss @KanchanChauhan will now test on our dev-system. Thank you for your input!

**Comment by wojosc** [comment]
from what I can see it works:


Filter on Task from Project
filter on timesheet record


Project fetched based on Task
filter on timesheet record

This also works. At the end of the GIF though you can see, that when we unselect the project there is a "Project = emtpy" Filter applied to Task. Can we fix this?

**Comment by KanchanChauhan** [comment]
@wojosc Created a PR to fix the empty project issue https://github.com/phamos-eu/Project-Addon/pull/23

**Pull request merged**